src/python/completions.rs

```diff
@@ -144,6 +144,104 @@ impl Backend {
 						}
 						Some(PyCompletions::Mapped) => {
 							if range.contains_end(offset) {
+								// Check if this is an ERROR node with broken syntax
+								tracing::debug!(
+									"Mapped capture node kind: {}, range: {:?}, offset: {}",
+									capture.node.kind(),
+									range,
+									offset
+								);
+								if capture.node.kind() == "ERROR" {
+									// This might be a string without a colon in a dictionary
+									let error_text = &contents[capture.node.byte_range()];
+									if error_text.starts_with(b"'") || error_text.starts_with(b"\"") {
+										// Extract the partial text
+										let quote_char = error_text[0];
+										let end_quote = error_text.iter().rposition(|&b| b == quote_char);
+										let needle = if let Some(end) = end_quote {
+											String::from_utf8_lossy(&error_text[1..end])
+										} else if offset > capture.node.start_byte() + 1 {
+											String::from_utf8_lossy(&error_text[1..offset - capture.node.start_byte()])
+										} else {
+											Cow::Borrowed("")
+										};
+
+										// Try to determine the model from context
+										// Look for the parent dictionary and field assignment
+										let mut current = capture.node;
+										let mut field_model = None;
+
+										while let Some(parent) = current.parent() {
+											if parent.kind() == "dictionary" {
+												// Found the dictionary, now look for the field assignment
+												if let Some(list_parent) = parent.parent()
+													&& list_parent.kind() == "list" && let Some(pair_parent) =
+													list_parent.parent() && pair_parent.kind() == "pair"
+													&& let Some(key) = pair_parent.child_by_field_name("key")
+													&& key.kind() == "string"
+												{
+													let field_name =
+														String::from_utf8_lossy(&contents[key.byte_range().shrink(1)]);
+													// Use the field name to determine model
+													if field_name.ends_with("_ids") || field_name.ends_with("_id") {
+														let hint =
+															if let Some(stripped) = field_name.strip_suffix("_ids") {
+																stripped
+															} else {
+																field_name.strip_suffix("_id").unwrap()
+															};
+
+														field_model = self
+															.index
+															.models
+															.iter()
+															.find(|entry| {
+																let model_name = _R(*entry.key());
+																model_name.ends_with(&format!(".{hint}"))
+																	|| (model_name.contains('.')
+																		&& model_name.split('.').next_back()
+																			== Some(hint))
+															})
+															.map(|entry| *entry.key());
+													}
+												}
+												break;
```

Comment:
It might be better to reuse the logic to determine the chain of accesses `foo.bar.baz` as in gather_commandlist here, as there are not too many fields that would fit into this hint